### DISCUSSION TITLE
Gestures to move in 2d/3d viewport in touch editor

### DISCUSSION THREAD

**Discussion by GumaD3v** [godotengine/godot-proposals#5392] [Editor]
Rn the Godot Editor is unusable for various reasons when you don't have mouse and keyboard, one of them is the impossibility of moviment in 3d, in 2d is possible(with the sliders), but is uncomfortable to use, it would be a step forward to make the Godot Editor in Godot implementing moviment with gestures like:

EDIT: as @Calinou commented, I'm marking the features already targeted in [https://github.com/godotengine/godot-proposals/issues/5124](#5124)

3D:
- Rotate with swipe - already in feature request
- Move (like in blender when you press Shift+MiddleMouseButton) with 2 finger swipe - already in feature request

2D:
- Move with swipe

General:
- Pinch to zoom - already in feature request
- Selection box with hold+swipe - already in feature request
- Undo with three finger swipe to the left
- Redo with three finger swipe to the right

**Comment by Calinou** [comment]
3D editor controls are already being tracked in https://github.com/godotengine/godot-proposals/issues/5124.

**Comment by GumaD3v** [comment]
I didn't know, however it fixes half what i wrote, so I'm marking the ones that are already present in the request you mentioned but keeping the others
